common/djangoapps/track/views/segmentio.py

```diff
@@ -158,8 +159,14 @@ def track_segmentio_event(request):  # pylint: disable=too-many-statements
     if 'data' not in segment_properties:
         raise EventValidationError(ERROR_MISSING_DATA)
 
+    set_custom_attribute('segment_event_name', segment_event_name)
+    set_custom_attribute('segment_event_source', event_source)
+
     # create and populate application field if it doesn't exist
     app_context = segment_properties.get('context', {})
+    if type(app_context) is not dict:
+        set_custom_attribute('segment_unexpected_context', str(app_context))
+        app_context = {}
```

Comment:
**Reviewer Note**
1. Someone could add a test.
2. Could log and/or raise an exception as well if that is more appropriate.

Reply:
@brianhw: It sounds like this might have covered up the potential Mobile issue, so maybe leave the custom attribute and then raise an exception?  Anyway, your team can decide what would be best.  

The custom attributes will help answer some of your questions via New Relic, like the source, etc.